crates/forge/src/compatibility_check.rs

```diff
@@ -105,11 +125,13 @@ pub fn create_version_parser<'a>(name: &'a str, pattern: &'a str) -> Box<Version
     })
 }
 
-fn get_raw_version(name: &str, command: &RefCell<Command>) -> Result<String> {
-    let raw_current_version = command
-        .borrow_mut()
-        .output()
-        .with_context(|| format!("Failed to run version command for {name}"))?;
+fn get_version(command: &RefCell<Command>, parser: &VersionParser) -> Result<Version> {
+    let raw_version = get_raw_version(command)?;
+    parser(&raw_version)
+}
+
+fn get_raw_version(command: &RefCell<Command>) -> Result<String> {
+    let raw_current_version = command.borrow_mut().output_checked()?;
```

Comment:
1. Let's inline `get_raw_version`, it's very short and used just once (inside `get_version`):

```rust
fn get_version(command: &RefCell<Command>, parser: &VersionParser) -> Result<Version> {
    let raw_current_version = command.borrow_mut().output_checked()?;
    let raw_version = String::from_utf8_lossy(&raw_current_version.stdout)
        .trim()
        .to_string();
    parser(&raw_version)
}
```

2. `get_version` should be a method of `Requirement`.